WordPress/Classes/Models/Notifications/NotificationSettings.swift

```diff
@@ -0,0 +1,185 @@
+        case .WordPressCom:
+            return wpcomPreferenceKeys
+        }
+    }
+    
+
+    /**
+    *  @enum        Channel
+    *  @brief       Represents a communication channel that may post notifications to the user.
+    */
+    public enum Channel : Equatable {
+        case Blog(blogId: Int)
+        case Other
+        case WordPressCom
+        
+        
+        /**
+        *  @details Returns the localized description of the current enum value
+        */
+        func description() -> String {
+            switch self {
+            case .Blog:
+                return NSLocalizedString("WordPress Blog", comment: "Settings for a Wordpress Blog")
+            case .Other:
+                return NSLocalizedString("Comments on Other Sites", comment: "Notification Settings Channel")
+            case .WordPressCom:
+                return NSLocalizedString("Updates from WordPress.com", comment: "Notification Settings Channel")
+            }
+        }
+    }
+    
+    
+    /**
+    *  @class       Stream
+    *  @brief       Contains the Notification Settings collection for a specific communications stream.
+    */
+    public class Stream {
+        public var kind         : Kind
+        public var preferences  : [String : Bool]?
+        
+        
+        /**
+        *  @details Designated Initializer
+        *  @param   kind            The Kind of stream we're currently dealing with
+        *  @param   preferences     Raw remote preferences, retrieved from the backend
+        */
+        public init(kind: String, preferences: [String : Bool]?) {
+            self.kind           = Kind(rawValue: kind) ?? .Email
+            self.preferences    = preferences
+        }
+        
+        
+        /**
+        *  @enum    Stream.Kind
+        *  @brief   Enumerates all of the possible Stream Kinds
+        */
+        public enum Kind : String {
+            case Timeline       = "timeline"
+            case Email          = "email"
+            case Device         = "device"
+            
+            
+            /**
+            *  @details Returns the localized description of the current enum value
+            */
+            func description() -> String {
+                switch self {
+                case .Timeline:
+                    return NSLocalizedString("Notifications Tab", comment: "WordPress.com Notifications Timeline")
+                case .Email:
+                    return NSLocalizedString("Email", comment: "Email Notifications Channel")
+                case .Device:
+                    return NSLocalizedString("Push Notifications", comment: "Mobile Push Notifications")
+                }
+            }
+            
+            static let allValues = [ Timeline, Email, Device ]
+        }
+    }
+    
+    // MARK: - Private Properties
+    private let blogPreferenceKeys      = ["new_comment", "comment_like", "post_like", "follow", "achievement", "mentions"]
```

Comment:
Yes I know it's ironic that I'm saying this ... but... should these values be constants defined somewhere? Other than preventing spelling mistakes in the keys I'm not sure if they're useful in the UI or not.